src/biginteger.ts

```diff
@@ -0,0 +1,499 @@
+/**
+ * @fileoverview
+ * BigInteger implementation of basic operations
+ * that wraps the native BigInt library.
+ * Operations are not constant time,
+ * but we try and limit timing leakage where we can
+ */
+export default class BigInteger {
```

Comment:
Do we still need this wrapper class? Perhaps we can just replace the uses of this class with direct uses of the operators (and perhaps a few helper functions) everywhere?